core/src/main/scala/cats/data/ReaderWriterStateT.scala

```diff
@@ -1,59 +1,84 @@
 package cats
 package data
 
-import cats.functor.{ Contravariant, Profunctor }
+import cats.functor.{ Bifunctor, Contravariant, Profunctor }
 import cats.syntax.either._
 
 /**
- * Represents a stateful computation in a context `F[_]`, over state `S`, with an initial environment `E`,
- * an accumulated log `L` and a result `A`.
+ * Represents a stateful computation in a context `F[_]`, from state `SA` to state `SB`,
+ *  with an initial environment `E`, an accumulated log `L` and a result `A`.
  *
  * In other words, it is a pre-baked stack of `[[ReaderT]][F, E, A]`, `[[WriterT]][F, L, A]`
- * and `[[StateT]][F, S, A]`.
+ * and `IndexedStateT[F, SA, SB, A]`.
  */
-final class ReaderWriterStateT[F[_], E, L, S, A](val runF: F[(E, S) => F[(L, S, A)]]) extends Serializable {
+final class IndexedReaderWriterStateT[F[_], E, L, SA, SB, A](val runF: F[(E, SA) => F[(L, SB, A)]]) extends Serializable {
```

Comment:
it feels like this could have an `Arrow[SA, SB]` instance now. Am I wrong?

I don't know if it is worth it, but it if it can have an instance, it would be nice to add (same may be true for `IndexedStateT`).

Reply:
Hmm, we can definitely do `Compose` and `Strong`. 

`Category` (`id`) and `Arrow` (`lift`) are a bit more problematic. We need to conjure up the value somehow. That could work for `V: Monoid`.

I'll add `Compose`+`Strong` and `Arrow` as separate traits. Thanks for the suggestion

Reply:
@johnynek Added the instances both here and in the IndexedStateT PR.